src/cli/grain.js

```diff
@@ -73,6 +78,64 @@ const grainCommand: Command = async (args, std) => {
   return 0;
 };
 
+/**
+ * Merge receipts under the same name into one.
+ */
+function mergeReceipts(
+  distributions: $ReadOnlyArray<Distribution>
+): $ReadOnlyArray<GrainReceipt> {
+  const mergedBalances: {[string]: G.Grain} = {};
+  for (const {allocations} of distributions) {
+    for (const {receipts} of allocations) {
+      for (const {amount, id} of receipts) {
+        if (!mergedBalances[id.toString()]) {
+          mergedBalances[id.toString()] = amount;
+        } else {
+          const existingBalance = mergedBalances[id.toString()];
+          mergedBalances[id.toString()] = G.add(amount, existingBalance);
+        }
+      }
```

Comment:
If I were writing this I would do:

```js
const balances: Map<IdentityId, Grain> = new Map();
for (const {amount, id} of stuff) {
  const existingBalance = NullUtil.orElse(balances.get(id), G.ZERO);
  balances.set(id, G.add(amount, existingBalance));
}
```

I think it's a cleaner pattern to avoid the if casing and to have only a single assign. Across the codebase, we don't use objects as maps, preferring ES6 maps. Unlike objects, they are well defined for all string inputs. cc @wchargin